velox/common/config/SpillConfig.h

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+#include <stdint.h>
+#include <string.h>
+
+#include <folly/executors/CPUThreadPoolExecutor.h>
+#include "velox/common/compression/Compression.h"
+
+namespace facebook::velox::common {
+// Specifies the config for spilling.
+struct SpillConfig {
+  SpillConfig(
+      const std::string& _filePath,
+      uint64_t _maxFileSize,
+      uint64_t _minSpillRunSize,
+      folly::Executor* _executor,
+      int32_t _spillableReservationGrowthPct,
+      uint8_t _startPartitionBit,
+      uint8_t _joinPartitionBits,
+      uint8_t _aggregationPartitionBits,
+      int32_t _maxSpillLevel,
+      int32_t _testSpillPct,
+      const std::string& _compressionKind)
+      : filePath(_filePath),
+        maxFileSize(
+            _maxFileSize == 0 ? std::numeric_limits<int64_t>::max()
+                              : _maxFileSize),
+        minSpillRunSize(_minSpillRunSize),
+        executor(_executor),
+        spillableReservationGrowthPct(_spillableReservationGrowthPct),
+        startPartitionBit(_startPartitionBit),
+        joinPartitionBits(_joinPartitionBits),
+        aggregationPartitionBits(_aggregationPartitionBits),
+        maxSpillLevel(_maxSpillLevel),
+        testSpillPct(_testSpillPct),
+        compressionKind(common::stringToCompressionKind(_compressionKind)) {}
+
+  /// Returns the hash join spilling level with given 'startBitOffset'.
+  ///
+  /// NOTE: we advance (or right shift) the partition bit offset when goes to
+  /// the next level of recursive spilling.
+  int32_t joinSpillLevel(uint8_t startBitOffset) const;
```

Comment:
Can you add SpillConfigTest.cpp to move the current tests on these APIs from SpillerTest.cpp? Thanks!